lib/cloud-operation/communication/websocket-channel.ts

```diff
@@ -0,0 +1,76 @@
+import { v4 } from "uuid";
+
+import { CommunicationChannelBase } from "./communication-channel-base";
+import { CloudOperationWebSocketMessageActions, CloudCommunicationEvents, CloudCommunicationChannelExitCodes } from "../../constants";
+
+export class WebSocketCommunicationChannel extends CommunicationChannelBase {
+	private ws: IWebSocket;
+
+	constructor(protected cloudOperationId: string,
+		protected data: ICloudCommunicationChannelData<IWebSocketCloudChannelConfigProperties>,
+		protected $logger: ILogger,
+		private $nsCloudWebSocketFactory: IWebSocketFactory) {
+		super(cloudOperationId, data, $logger);
+	}
+
+	protected async closeCore(code: number, reason?: string): Promise<void> {
+		if (this.ws.readyState === this.ws.OPEN) {
+			this.ws.removeAllListeners();
```

Comment:
I think you should move this outside of the IF - e.g. if connectCore fails, who will remove the open handler.

Reply:
You are correct.